src/lib/autoinstall/clients/autoinst_files_finish.rb

```diff
@@ -0,0 +1,18 @@
+require "installation/finish_client"
+
+Yast.import "AutoinstFile"
+
+module Y2Autoinstallation
+  module Clients
+    class AutoinstFilesFinish < ::Installation::FinishClient
+      def title
+        textdomain "autoinst"
+        _("Writing configuration files ...")
+      end
+
+      def write
+        ::Yast::AutoinstFile.Write
+      end
```

Comment:
I would define also mode, so it is not run always, just in autoyast. See https://github.com/yast/yast-yast2/blob/30a9253009a02a8fb0366c372fe82cdb746dfd40/library/general/src/lib/installation/finish_client.rb#L72